### PR TITLE
Getting rid of BYTE type (#23)

### PR DESCRIPTION
**Pull request by Plotja**
#23

### CODE REVIEW

**Comment by nitrocaster** [comment]
Please, use `git commit --amend` and `git push --force` to avoid creating all these 'fix' commits. Forced pushes are OK when you're committing to your own temporary branch.

**Comment by Plotja** [comment]
> Please, use `git commit --amend` and `git push --force` to avoid creating all these 'fix' commits. Forced pushes are OK when you're committing to your own temporary branch.

You are right. I fixed this.

**Comment by LVutner** [comment]
Soo, what's the problem with BYTE? There's any point in replacing it?

**Comment by ghost** [comment]
If a BYTE is assigned there, why do you change it? Then - oops have us crashes(((
The same goes for the IC if it has a define assigned to it. So don't touch it :ritas:

**Comment by Plotja** [comment]
> Soo, what's the problem with BYTE? There's any point in replacing it?

Some problems with portability.

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]
Please, revert `utils/mp_gpprof_server`.

File: src/utils/mp_gpprof_server/libraries/gamespy/gt2/gt2action/TGAFile.cpp [hunk 1]
```diff
@@ -1571,4 +1571,4 @@ extern "C" unsigned char * LoadTGAFile
 {
 	TGAFile tgaFile;
 	return (unsigned char *)tgaFile.LoadTGA(filename, (UINT32 *)width, (UINT32 *)height);
-}
\ No newline at end of file
+}
```
Comment (Xottab-DUTY) [inline src/utils/mp_gpprof_server/libraries/gamespy/gt2/gt2action/TGAFile.cpp]: Remove the empty line, please. We don't want to produce the diff for no reason. :)

File: src/xrGame/RegistryFuncs.cpp [hunk 2]
```diff
@@ -96,7 +96,7 @@ bool WriteRegistryValue(LPCSTR rKeyName, DWORD rKeyType, const void* value)
     break;
     };
 
-    res = RegSetValueEx(hKey, rKeyName, NULL, rKeyType, (LPBYTE)value, KeyValueSize);
+    res = RegSetValueEx(hKey, rKeyName, NULL, rKeyType, (u8*)value, KeyValueSize);
```
Comment (Xottab-DUTY) [inline src/xrGame/RegistryFuncs.cpp]: Let's use LPBYTE when interacting with WinAPI.

File: src/xrGameSpy/xrGameSpy.cpp [hunk 3]
```diff
@@ -38,7 +38,7 @@ int GetGameDistribution()
     if (res == ERROR_SUCCESS && KeyCDKey != 0)
     {
         res = RegQueryValueEx(
-            KeyCDKey, REGISTRY_VALUE_INSTALL_PATCH_ID, NULL, &KeyValueType, (LPBYTE)&KeyValue, &KeyValueSize);
+            KeyCDKey, REGISTRY_VALUE_INSTALL_PATCH_ID, NULL, &KeyValueType, (u8*)&KeyValue, &KeyValueSize);
```
Comment (Xottab-DUTY) [inline src/xrGameSpy/xrGameSpy.cpp]: Let's use LPBYTE when interacting with WinAPI.

File: src/xrGame/RegistryFuncs.cpp [hunk 4]
```diff
@@ -42,7 +42,7 @@ bool ReadRegistryValue(LPCSTR rKeyName, DWORD rKeyType, void* value)
     break;
     };
 
-    res = RegQueryValueEx(hKey, rKeyName, NULL, &rKeyType, (LPBYTE)rBuf, &KeyValueSize);
+    res = RegQueryValueEx(hKey, rKeyName, NULL, &rKeyType, (u8*)rBuf, &KeyValueSize);
```
Comment (Xottab-DUTY) [inline src/xrGame/RegistryFuncs.cpp]: Let's use LPBYTE when interacting with WinAPI.

**Review by Xottab-DUTY — APPROVED** [review]

**Pull request merged**